### PR TITLE
[request] Wallabag

### PR DESCRIPTION
**Pull request by tolkonepiu**
- Website: https://wallabag.org/en
- Version: 2.2.3
- Description: Wallabag is a self hostable application allowing you to not miss any content anymore. Click, save, read it when you can. It extracts content so that you can read it when you have time.
- Documentation: https://doc.wallabag.org/en/
- Source code: https://github.com/wallabag/wallabag
- License: MIT

### CODE REVIEW

**Comment by cytec** [comment]
looks good to me \cc @Dr-Bean @Diaoul

File: spk/wallabag/Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+SPK_NAME = wallabag
+SPK_VERS = 1.9.1-b
```
Comment (Diaoul) [inline spk/wallabag/Makefile]: We need to be careful with letters and special characters in version number, the version comparison in DSM is somewhat tricky and with this 1.9.2 could be considered a lower version than 1.9.1-b.
I had some problems with that in the past

**Comment by Diaoul** [comment]
LGTM, please check the update mechanism though.

File: spk/wallabag/src/installer.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Install the web interface
+    cp -R ${INSTALL_DIR}/share/${PACKAGE} ${WEB_DIR}
+
+    # permissions
+    chown -R ${USER} ${WEB_DIR}/${SHORTNAME}
```
Comment (Dr-Bean) [inline spk/wallabag/src/installer.sh]: Probably should be changed to `${WEB_DIR}/${PACKAGE}`

File: spk/wallabag/Makefile (hunk 1 above)
Comment (tolkonepiu) [inline spk/wallabag/Makefile]: How it works?

Comment (cytec) [inline spk/wallabag/Makefile]: just remove the -b thing ...
change the line to `SPK_VERS = 1.9.1`

Comment (tolkonepiu) [inline spk/wallabag/Makefile]: ok

**Comment by cytec** [comment]
looks good to me now \cc @SynoCommunity/developers

File: spk/wallabag/src/installer.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    # Remove the web interface
+    rm -fr ${WEB_DIR}/${PACKAGE}
+
+    exit 0
+}
+
+preupgrade ()
+{	
+	rm -fr ${WEB_DIR}/${PACKAGE}/cache/*
```
Comment (Dr-Bean) [inline spk/wallabag/src/installer.sh]: Replace tabs with spaces in this file please. Seems to be limited to the later *upgrade stuff.

File: spk/wallabag/src/installer.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	rm -fr ${WEB_DIR}/${PACKAGE}/install/
```
Comment (Dr-Bean) [inline spk/wallabag/src/installer.sh]: L51 and L52 should not be necessary, as you presumably overwrite the contents via https://github.com/SynoCommunity/spksrc/pull/2165/files#diff-c0d645b343ac3265489c0734c80f5452R25.
Please verify and correct accordingly.

Comment (tolkonepiu) [inline spk/wallabag/src/installer.sh]: http://doc.wallabag.org/en/stable/Administrator/update.html

File: cross/wallabag/digests [hunk 5]
```diff
@@ -0,0 +1,3 @@
+1.9.1-b.zip SHA1 48E81C712008A00B9BC8B89ABEB1A135DE2BA2DD
```
Comment (Dr-Bean) [inline cross/wallabag/digests]: Please set `PKG_DIST_FILE = $(PKG_NAME)-$(PKG_VERS)` in `cross/wallabag/Makefile`. Then redownload and recreate digests. This ensures the downloaded file in $distrib is named `wallabag-1.9.1-b.zip`, making it easier to identify.

File: spk/wallabag/Makefile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+DEPENDS  = cross/$(SPK_NAME)
+
+MAINTAINER = tolkonepiu
+DESCRIPTION = Wallabag is a self hostable application allowing you to not miss any content anymore. Click, save, read it when you can. It extracts content so that you can read it when you have time.
+ADMIN_URL = /wallabag/
+RELOAD_UI = yes
+STARTABLE = no
+DISPLAY_NAME = Wallabag
+CHANGELOG = "Initial release"
```
Comment (Dr-Bean) [inline spk/wallabag/Makefile]: Please add `BETA = 1` after this, standard practice for new packages.

File: spk/wallabag/src/installer.sh [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    rm -fr ${TMP_DIR}/${PACKAGE}
+    mkdir -p ${TMP_DIR}/${PACKAGE}
+    mv ${WEB_DIR}/${PACKAGE}/inc/poche/config.inc.php ${TMP_DIR}/${PACKAGE}/
+    mv ${WEB_DIR}/${PACKAGE}/db ${TMP_DIR}/${PACKAGE}/
+    exit 0
+}
+
+postupgrade ()
+{
+    rm -fr ${WEB_DIR}/${PACKAGE}/cache/*
```
Comment (Dr-Bean) [inline spk/wallabag/src/installer.sh]: I'm reasonably sure these two entries (this and next line) are simply unnecessary. After an upgrade, there should not be any content in `cache` and `install`, so clearing it won't do anything.

**Comment by Dr-Bean** [comment]
Couple more comments, should be good to go with that I think.

File: spk/wallabag/src/installer.sh (hunk 7 above)
Comment (tolkonepiu) [inline spk/wallabag/src/installer.sh]: Ok

File: cross/wallabag/Makefile [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+PKG_VERS = 1.9.1-b
+PKG_EXT = zip
+PKG_DIST_NAME = $(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/wallabag/wallabag/archive
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS =
+
+HOMEPAGE = https://www.wallabag.org/
+COMMENT  = Wallabag is a self hostable application allowing you to not miss any content anymore. Click, save, read it when you can. It extracts content so that you can read it when you have time.
```
Comment (0wnrepo) [inline cross/wallabag/Makefile]: ...allowing you to save an offline copy of your favorite articles. 

sounds a bit better, don't you think ?

File: spk/wallabag/Makefile (hunk 6 above)
Comment (tolkonepiu) [inline spk/wallabag/Makefile]: https://github.com/SynoCommunity/spksrc/pull/2165/commits/c1aef6fc6b2a9c12eac03e5d8e2c048f16da4051

File: cross/wallabag/Makefile (hunk 8 above)
Comment (tolkonepiu) [inline cross/wallabag/Makefile]: I took the description here: https://github.com/wallabag/wallabag 😄

File: spk/wallabag/src/installer.sh (hunk 7 above)
Comment (tolkonepiu) [inline spk/wallabag/src/installer.sh]: I'm not sure. 😞

Comment (tolkonepiu) [inline spk/wallabag/src/installer.sh]: https://github.com/wallabag/documentation/blob/master/en/Administrator/Update.md

**Comment by lennvilardi** [comment]
Any updates on version 2 ?

**Comment by tolkonepiu** [comment]
@lennvilardi, i plan to do it.

**Comment by nicosomb** [comment]
@tolkonepiu If you have any questions about wallabag installation, ask me ;-)

**Comment by tolkonepiu** [comment]
@cytec @Dr-Bean @Diaoul @nicosomb 
I just woke up: https://github.com/SynoCommunity/spksrc/pull/2165/commits/35476890a0b82f64ba25888b3adb9c5a3c91e569

**Comment by tolkonepiu** [comment]
@nicosomb can i do this https://github.com/SynoCommunity/spksrc/pull/2165/commits/024684c4cec2367b8a8d361c1d5c38320ef25c4b on 2.3.2?

**Comment by tolkonepiu** [comment]
https://github.com/SynoCommunity/spksrc/pull/2824 merged

**Pull request closed without merging**